syntax = "proto3";

package transcription;

service TranscriptionService {
  // Bidirectional streaming RPC for real-time transcription
  rpc StreamTranscription (stream AudioRequest) returns (stream TranscriptionResponse) {}
}

message AudioRequest {
  bytes audio_data = 1;
  string language = 2;
  string room_id = 3;
  string user_id = 4;
}

message TranscriptionResponse {
  string text = 1;
  float confidence = 2;
  string user_id = 3;
  string room_id = 4;
  bool is_final = 5;
  string error = 6;
} 